avocado/plugins/vmimage.py

```diff
@@ -0,0 +1,104 @@
+import os
+
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import CLICmd
+from avocado.core import data_dir, output
+from avocado.utils import vmimage, astring
+
+
+def list_downloaded_images():
+    """
+    List the available Image inside avocado cache
+
+    :return: list with image's parameters
+    :rtype: list of dicts
+    """
+    images = []
+    cache_dirs = data_dir.get_cache_dirs()
+    providers = [provider() for provider in vmimage.list_providers()]
+
+    for cache_dir in cache_dirs:
+        for root, _, files in os.walk(cache_dir):
+            if files:
+                for filename in files:
+                    for provider in providers:
+                        image = provider.get_image_parameters(filename)
+                        if image is not None:
+                            image["name"] = provider.name
+                            image["file"] = os.path.join(root, filename)
+                            images.append(image)
+                            break
+    return images
+
+
+def download_image(provider, version=None, arch=None):
+    """
+    Downloads image to the avocado cache if its not already there
+
+    :param provider: Name of the Image Provider, usually matches the distro name.
+    :type provider: str
+    :param version: (optional) Version of the system image.
+    :type version: str
+    :param arch: (optional) Architecture of the system image.
+    :type arch: str
+    """
+    cache_dirs = data_dir.get_cache_dirs()
+    image = vmimage.get(name=provider, version=version,
+                        arch=arch, cache_dir=cache_dirs)
+
+    LOG_UI.info("The image was downloaded:")
+    display_images_list([image])
```

Comment:
Line 58 below `says display_images_list()` is expecting a list of dicts. Is this line 50 even working?

Reply:
You are right, this is not working, I will write tests for catching those things. Thanks a lot.